nemo/export/trt_llm/converter/model_to_trt_llm_ckpt.py

```diff
@@ -238,6 +239,246 @@ def handle_model_level_weights(model, tp_idx: int, pp_idx: int):
     return weights_dict
 
 
+def _get_layer_index(split_key):
```

Comment:
There are too many duplicate code here. Can we reuse some of the existing functions and add parameters and code as needed?

Reply:
Can you please carefully review this code? To my knowledge this function is not duplicated anywhere

Reply:
1. Didn't mean specifically _get_layer_index function.
2. Also, I don't mean the exact copy of the code by duplicate but rather code with similar logic.

Reply:
The convention export currently uses ([see here](https://github.com/NVIDIA/NeMo/blob/7fee8e7a0f576317e4113cd58282fa833358c574/nemo/export/trt_llm/converter/utils.py#L179)) is to hardcode index 1. For Nemo models the layer number is not guaranteed (for instance if the module has been wrapped in autocast) so we have unique logic to determine the layer index.